Diagnostic/diagnostic.py

```diff
@@ -828,6 +848,21 @@ def start_mdsd():
 
                 # mdsd is now up for at least 30 seconds.
 
+                # Mitigate if memory leak is suspected.
+                mdsd_memory_leak_suspected, mdsd_memory_usage_in_MB = check_suspected_memory_leak(mdsd.pid)
+                if mdsd_memory_leak_suspected:
+                    memory_leak_msg = "Suspected mdsd memory leak (memory usage: {0}MB). Recycling mdsd to self-mitigate.".format(mdsd_memory_usage_in_MB)
+                    hutil.log(memory_leak_msg)
+                    # Add a telemetry as well. This might ding our enable success KPI,
+                    # but there doesn't seem to be any other way.
+                    waagent.AddExtensionEvent(name=hutil.get_name(),
+                                              op=ExtensionOperationType,
+                                              isSuccess=False,
+                                              version=hutil.get_extension_version(),
+                                              message=memory_leak_msg)
```

Comment:
Because setting "isSuccess=True" may incorrectly make our Enable success rate better...? Call me a purist, but I'd choose to get penalized on this, rather than make it look incorrectly better than the actual.

I thought about using another extension operation type with success=true, but didn't look more until your comment here. I don't think there's an extension op type corresponding to auto-restart, but the closest seems to be "HeartBeat". Let me use that here with success=true. Thanks.